test/Chainweb/Test/Pact/PactInProcApi.hs

```diff
@@ -8,170 +10,129 @@
+        $ schedule Sequential
+            [ validateTest reqQIO
+            , localTest reqQIO
+            ]
+    , withPact testMemPoolAccess $ \reqQIO ->
+        newBlockTest "new-block-0" reqQIO
+    , withPact testEmptyMemPool $ \reqQIO ->
+        newBlockTest "empty block tests" reqQIO
+    ]
+
+withPact :: MemPoolAccess -> (IO (TQueue RequestMsg) -> TestTree) -> TestTree
+withPact mempool f = withResource startPact stopPact $ f . fmap snd
+  where
+    startPact = do
+        mv <- newEmptyMVar
+        reqQ <- atomically newTQueue
+        a <- async (PS.initPactService testVersion cid logger reqQ mempool mv)
+        return (a, reqQ)
+
+    stopPact (a, reqQ) = do
+        sendCloseMsg reqQ
+        cancel a
+
+    logger = genericLogger Warn T.putStrLn
+    cid = someChainId testVersion
+
+newBlockTest :: String -> IO (TQueue RequestMsg) -> TestTree
+newBlockTest label reqIO = pactGolden label $ do
+    reqQ <- reqIO
+    let genesisHeader = genesisBlockHeader testVersion cid
+    respVar <- newBlock noMiner genesisHeader reqQ
+    goldenBytes "new-block" =<< takeMVar respVar
+  where
+    cid = someChainId testVersion
+
+validateTest :: IO (TQueue RequestMsg) -> ScheduledTest
+validateTest reqIO = pactGoldenSch "validateBlock-0" $ do
+    reqQ <- reqIO
+    let genesisHeader = genesisBlockHeader testVersion cid
+    respVar0 <- newBlock noMiner genesisHeader reqQ
+    plwo <- takeMVar respVar0 >>= \case
+        Left e -> assertFailure (show e)
+        Right r -> return r
+
+    -- validate the same transactions sent to newBlockTest above
+    let matchingPlHash = _payloadWithOutputsPayloadHash plwo
+    let plData = PayloadData
+            { _payloadDataTransactions = fst <$> _payloadWithOutputsTransactions plwo
+            , _payloadDataMiner = _payloadWithOutputsMiner plwo
+            , _payloadDataPayloadHash = matchingPlHash
+            , _payloadDataTransactionsHash = _payloadWithOutputsTransactionsHash plwo
+            , _payloadDataOutputsHash = _payloadWithOutputsOutputsHash plwo
+            }
+
+    let headers = V.fromList $ getBlockHeaders cid 2
+    let toValidateHeader = (headers ! 1)
+            { _blockPayloadHash = matchingPlHash
+            , _blockParent = _blockHash genesisHeader
+            }
+    respVar1 <- validateBlock toValidateHeader plData reqQ
+    goldenBytes "validateBlock-0" =<< takeMVar respVar1
+  where
+    cid = someChainId testVersion
+
+localTest :: IO (TQueue RequestMsg) -> ScheduledTest
+localTest reqIO = pactGoldenSch "local" $ do
+    reqQ <- reqIO
+    locVar0c <- testLocal >>= \t -> local t reqQ
+    goldenBytes "local" =<< takeMVar locVar0c
+
+goldenBytes :: Y.ToJSON a => Exception e => String -> Either e a -> IO BL.ByteString
+goldenBytes label (Left e) = assertFailure $ label ++ ": " ++ show e
+goldenBytes label (Right a) = return $ BL.fromStrict $ Y.encode $ object
+    [ "test-group" .= label
+    , "results" .= a
+    ]
```

Comment:
See comment above. Usage of non-normalized Yaml/JSON for golden file encodings can cause flaky test results.

Reply:
I think the use of encode is not limited to the tests -- if it needs changing we'll have to find all the occurrences using in the various places conversions are done.

Reply:
My understanding is that these tests are not concerned with cryptographic consistency of the data but changes in the APIs. Since the different JSON/YAML encodings are semantically equal, the tests would still be valid. 

It's just about spurious test failures. So we may just move on and add a normalized encoding down the road if needed.

Reply:
If the tests were concerned about the cryptographic identity of the the data base the test should just compare the Merkle roots for the respective payload data structures.